packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -336,14 +336,40 @@ export default function useOnBlockDrop(
 
 	const moveBlocks = useCallback(
 		( sourceClientIds, sourceRootClientId, insertIndex ) => {
+			const sourceBlocks = getBlocksByClientId( sourceClientIds );
+
+			// TODO: Figure out a better place for this. useOnBlockDrop shouldn't care about layout styles.
```

Comment:
I'm struggling to come up with a better place/approach for this. Any ideas @tellthemachines?

Reply:
I wouldn't worry about contaminating drag and drop with layout logic; `useOnBlockDrop` [already cares about layout styles](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/use-on-block-drop/index.js#L291) 😅 

Drag and drop doesn't exist in a vacuum. It's an interaction that's tied to an editable canvas, and it makes sense that its possibilities are determined by layout, block types and block state. 

Would it be worth checking that there's a pin in place before running `updateBlockAttributes` on all the blocks being moved?